### ISSUE TITLE
MDiscussion - feedback from free label annotationn

### ISSUE THREAD

**Issue by dobkeratops** [bbernhard/imagemonkey-core#259]
Ok the unified mode with ability to annotate free labels is another huge step forward; 

Now we have the dilemma over what to do about label explosion.
Perhaps we could grow the graph I parallel with annotating to keep it under control..

I’ll pause a bit because it’s clear with free labelling , label explosion is a risk - i think you had this problem trying to go back over the LabelMe dataset

Some points:

- i definitely like being able to put a material prefix (metal plastic glass wooden ...) .. eg “metal gate, metal frame” etcmost of the time I think we can look for these words and just strip it from the front and turn it into a property ( a material “detector” network is a very likely use case for graphics .. imagine being able to assign materials to photogrammetry scans..)

- one strange example was “porcelain rabbit” .. this could also have been descrbed as a “figurine/rabbit” .. in this case the natural language description makes it clear it’s a figurine . This would be a case for an explicit alias (ie porcelain rabbit would be an example of porcelain object, and figurine, and rabbit shape)

- Theres also the prefix “derelict” , again I think that can be extracted ok ( derelict building , derelict car, derelict aircraft... )

- Previously added a few posture or action labels for people, separated using the `/` .. it might be handy if searches for any specific label also returned combinations . Unable to use the `/` in Unified mode I’ve made a few explicit labels like “man sitting” as a workaround. I’m hoping aliases  can catch everything eventually . Again it might be possible that a few words like “sitting standing walking” etc can be separated unambiguously 

- I started using “or”,”of”,”and” to replace `/` for combinations and parts .. “head of man” “trees and bushes”

- explicit plurals: “trees” and “buildings” are handy to have since these are very rarely isolated ; also “people” and “crowd” ... crowd implies both plural person but also high density 

- there were some more complicated examples which could only be handled by aliases I think , eg “plastic food box” ( a certain type of plastic box), “plate of food” seemed like a useful compound label (think of the meal analyser app .. a thing to zero in on) “glass of water” / “empty glass” “glass cup” ( I like these because they disambiguate the word glass)

**Comment by bbernhard** [comment]
> i definitely like being able to put a material prefix (metal plastic glass wooden ...) .. eg “metal gate, wooden chair, metal frame” etcmost of the time I think we can look for these words and just strip it from the front and turn it into a property ( a material “detector” network is a very likely use case for graphics .. imagine being able to assign materials to photogrammetry scans..). Some generic object words like box, panel,frame would allow picking out a lot of material examples

agreed :) I think that's something we can (easily) automate. There are probably some edge cases where we can't automatically parse the label (e.g ambiguity), but in those (rare) cases a user/admin can help the system to resolve the ambiguity. 

> one strange example was “porcelain rabbit” .. this could also have been descrbed as a “figurine/rabbit”. This would be a case for an explicit alias

> Theres also the prefix “derelict” , again I think that can be extracted ok ( derelict building , derelict car, derelict aircraft... )


aliases and "prefix parsing" are the next two big topics on my todo list (after the server move). I think those two mechanisms could really help us to control the label explosion. (I'll later create a new discussion thread, to brainstorm a bit together about the technical details/requirements of those new features). 


> I started using “or”,”of”,”and” to replace / for combinations and parts .. “head of man” “trees and bushes”

perfect! :+1: 

> I’ve been able to give a few examples of urban street layout features : central reservation, traffic island, pedestrian crossings .. eventually these will be good labels for self driving car vocabulary

awesome, that's very cool! :)
